### PR TITLE
ISSUE-4345 integrate fuse table with new processor

### LINKED ISSUE #4345: integrate fuse table with new processor

### PR DESCRIPTION
**Pull request by zhang2014**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

integrate fuse table with new processor

## Changelog

- New Feature

## Related Issues

Fixes #4345

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

**Comment by zhang2014** [comment]
all table engines supported new processor after this pr

**Comment by BohuTANG** [comment]
:tada: 

I have test [ontime on s3](https://databend.rs/doc/performance/ec2-s3-performance), there is a bit performance improve.

File: query/src/storages/fuse/operations/read.rs [hunk 1]
```diff
@@ -32,6 +43,34 @@ impl FuseTable {
         ctx: Arc<QueryContext>,
         push_downs: &Option<Extras>,
     ) -> Result<SendableDataBlockStream> {
+        let block_reader = self.create_block_reader(&ctx, push_downs)?;
+
+        let bite_size = ctx.get_settings().get_parallel_read_threads()?;
+        // let ctx_clone = ctx.clone();
```
Comment (BohuTANG) [inline query/src/storages/fuse/operations/read.rs:49]: No used anymore?

**Comment by BohuTANG** [comment]
Performance:

main:
```
| Command | Mean [ms] | Min [ms] | Max [ms] | Relative |
|:---|---:|---:|---:|---:|
| `mysql --host 127.0.0.1 -P 3307 -uroot -s </tmp/bench_0.sql` | 304.9 ± 73.9 | 253.3 | 483.3 | 1.24 ± 0.31 |
| `mysql --host 127.0.0.1 -P 3307 -uroot -s </tmp/bench_1.sql` | 378.7 ± 83.7 | 329.0 | 601.3 | 1.54 ± 0.36 |
| `mysql --host 127.0.0.1 -P 3307 -uroot -s </tmp/bench_2.sql` | 423.8 ± 12.0 | 403.7 | 444.5 | 1.73 ± 0.14 |
| `mysql --host 127.0.0.1 -P 3307 -uroot -s </tmp/bench_3.sql` | 256.4 ± 71.0 | 208.3 | 447.0 | 1.04 ± 0.30 |
| `mysql --host 127.0.0.1 -P 3307 -uroot -s </tmp/bench_4.sql` | 245.4 ± 17.9 | 222.6 | 278.1 | 1.00 |
| `mysql --host 127.0.0.1 -P 3307 -uroot -s </tmp/bench_5.sql` | 477.8 ± 53.1 | 449.8 | 627.5 | 1.95 ± 0.26 |
| `mysql --host 127.0.0.1 -P 3307 -uroot -s </tmp/bench_6.sql` | 518.9 ± 123.0 | 451.3 | 826.1 | 2.11 ± 0.52 |
| `mysql --host 127.0.0.1 -P 3307 -uroot -s </tmp/bench_7.sql` | 714.2 ± 264.2 | 536.0 | 1420.2 | 2.91 ± 1.10 |
| `mysql --host 127.0.0.1 -P 3307 -uroot -s </tmp/bench_8.sql` | 406.6 ± 24.8 | 385.9 | 470.8 | 1.66 ± 0.16 |
| `mysql --host 127.0.0.1 -P 3307 -uroot -s </tmp/bench_9.sql` | 546.6 ± 72.9 | 506.9 | 747.9 | 2.23 ± 0.34 |
| `mysql --host 127.0.0.1 -P 3307 -uroot -s </tmp/bench_10.sql` | 543.5 ± 35.0 | 505.7 | 607.6 | 2.21 ± 0.22 |
| `mysql --host 127.0.0.1 -P 3307 -uroot -s </tmp/bench_11.sql` | 2934.0 ± 22.9 | 2894.6 | 2969.5 | 11.96 ± 0.88 |
| `mysql --host 127.0.0.1 -P 3307 -uroot -s </tmp/bench_12.sql` | 1248.1 ± 13.3 | 1225.3 | 1274.3 | 5.09 ± 0.38 |
```

This PR:
```
| Command | Mean [ms] | Min [ms] | Max [ms] | Relative |
|:---|---:|---:|---:|---:|
| `mysql --host 127.0.0.1 -P 3307 -uroot -s </tmp/bench_0.sql` | 278.2 ± 57.0 | 238.0 | 435.6 | 1.26 ± 0.26 |
| `mysql --host 127.0.0.1 -P 3307 -uroot -s </tmp/bench_1.sql` | 343.7 ± 63.0 | 302.0 | 474.6 | 1.56 ± 0.29 |
| `mysql --host 127.0.0.1 -P 3307 -uroot -s </tmp/bench_2.sql` | 412.6 ± 22.5 | 387.5 | 454.0 | 1.87 ± 0.13 |
| `mysql --host 127.0.0.1 -P 3307 -uroot -s </tmp/bench_3.sql` | 220.1 ± 9.0 | 205.7 | 232.8 | 1.00 |
| `mysql --host 127.0.0.1 -P 3307 -uroot -s </tmp/bench_4.sql` | 256.7 ± 34.7 | 214.3 | 328.1 | 1.17 ± 0.16 |
| `mysql --host 127.0.0.1 -P 3307 -uroot -s </tmp/bench_5.sql` | 456.5 ± 10.0 | 439.5 | 474.3 | 2.07 ± 0.10 |
| `mysql --host 127.0.0.1 -P 3307 -uroot -s </tmp/bench_6.sql` | 469.4 ± 56.2 | 433.6 | 624.0 | 2.13 ± 0.27 |
| `mysql --host 127.0.0.1 -P 3307 -uroot -s </tmp/bench_7.sql` | 602.0 ± 36.2 | 554.4 | 655.6 | 2.73 ± 0.20 |
| `mysql --host 127.0.0.1 -P 3307 -uroot -s </tmp/bench_8.sql` | 468.5 ± 22.0 | 451.6 | 528.5 | 2.13 ± 0.13 |
| `mysql --host 127.0.0.1 -P 3307 -uroot -s </tmp/bench_9.sql` | 574.3 ± 12.1 | 554.2 | 589.3 | 2.61 ± 0.12 |
| `mysql --host 127.0.0.1 -P 3307 -uroot -s </tmp/bench_10.sql` | 553.3 ± 27.3 | 521.4 | 607.2 | 2.51 ± 0.16 |
| `mysql --host 127.0.0.1 -P 3307 -uroot -s </tmp/bench_11.sql` | 2888.6 ± 35.4 | 2831.9 | 2945.6 | 13.12 ± 0.56 |
| `mysql --host 127.0.0.1 -P 3307 -uroot -s </tmp/bench_12.sql` | 1243.4 ± 38.4 | 1211.7 | 1346.7 | 5.65 ± 0.29 |

```

**Review by sundy-li — APPROVED** [review]

**Pull request merged**